components/button/button-add.js

```diff
@@ -59,30 +79,131 @@ class ButtonAdd extends PropertyRequiredMixin(FocusMixin(LocalizeCoreElement(Lit
 			button:${unsafeCSS(getFocusPseudoClass())} .line {
 				border-top-color: var(--d2l-color-celestine);
 			}
+		`;
+	}
 
-			.content {
+	constructor() {
+		super();
+
+		this.hintNearbyDistance = DEFAULT_HINT_DISTANCE;
+		this.hintTiming = HINT_TIMING.ALWAYS;
+		this.textVisible = false;
+
+		this._buttonId = getUniqueId();
+		this._hasFocusOrHover = false;
+		this._visibleDistance = 0;
+
+		this.addEventListener('focus', this._onFocusHover);
+		this.addEventListener('mouseover', this._onFocusHover);
+		this.addEventListener('blur', this._onBlur);
+		this.addEventListener('mouseout', this._onBlur);
+	}
+
+	static get focusElementSelector() {
+		return 'button';
+	}
+
+	render() {
+		const buttonSpacing = this._getButtonSpacing();
+		const text = this.text || this.localize('components.button-add.addItem');
+		const content = this.textVisible ? this._renderWithTextVisible(text) : this._renderWithTextHidden(text);
```

Comment:
I divided render out into text visible vs hidden since they are fairly different in what they do